File: ITP1/2/d.hs
import Control.Applicative

main = do
  [w, h, x, y, r] <- map ( read :: String -> Int ) . words <$> getLine
  let horizontal = r <= x && x <= w-r
  let vertical = r <= y && y <= h-r
  let result  | horizontal && vertical = "Yes"
              | otherwise = "No"
  putStrLn $ result

File: ITP1/2/d.cpp
#include <iostream>
using namespace std;


int main(){
    int W, H, x, y, r;
    bool horizontal, vertical;
    cin >> W >> H >> x >> y >> r;
    horizontal = r <= x && x <= W-r;
    vertical = r <= y && y <= H-r;
    if (horizontal && vertical) {
        cout << "Yes" << endl;
    } else {
        cout << "No" << endl;
    }
    return 0;
}
